lock-api/src/main/java/com/palantir/lock/client/UnreliableTimeLockService.java

```diff
@@ -31,30 +32,52 @@
 import com.palantir.logsafe.SafeArg;
 import com.palantir.logsafe.logger.SafeLogger;
 import com.palantir.logsafe.logger.SafeLoggerFactory;
+import com.palantir.timestamp.TimestampManagementService;
 import com.palantir.timestamp.TimestampRange;
+import java.security.SecureRandom;
 import java.util.List;
 import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 /**
  * A version of timelock service that has a chance to randomly lose locks during refresh or immediately after
  * acquiring. This is useful for testing the behavior of clients when locks are lost.
  */
 public final class UnreliableTimeLockService implements TimelockService {
+    // A partition is sized at 2^23 timestamps, and we want to skip through potentially many at a time.
+    private static final long THOUSAND_COARSE_PARTITIONS_SIZE = 1000 * (1L << 23);
+    private static final SecureRandom SECURE_RANDOM = DefaultNativeSamplingSecureRandomFactory.INSTANCE.create();
```

Comment:
the name should reflect that this is fuzzer-controlled / or this should be spun out to a its own class/abstraction